### PR TITLE
audio/mp3, audio/vorbis, audio/wav: add Resample

### PR DESCRIPTION
**Pull request by RusAD32**
updates (and probably closes) #2055

### CODE REVIEW

**Comment by hajimehoshi** [comment]
Thanks, but before adding a public API, we should have a discussion.

For resampling, I want an API at the audio package rather than each package audio/mp3, audio/vorbis, and audio/wav.

**Comment by hajimehoshi** [comment]
> should the function operate on a general io.Reader

Yes it is. We want this function:

```go
package audio

func Resample(source io.ReadSeeker, size int64, from, to int) io.ReadSeeker
```

The implementation is already in audio/internal/convert.

**Comment by RusAD32** [comment]
Then it's basically a `convert.NewResampling` wrapper

File: audio/audio.go [hunk 1]
```diff
@@ -435,3 +436,13 @@ func (h *hookImpl) OnResumeAudio(f func() error) {
 func (h *hookImpl) AppendHookOnBeforeUpdate(f func() error) {
 	hooks.AppendHookOnBeforeUpdate(f)
 }
+
+// Resample explicitly resamples a stream to fit a new sample rate
+//
+// if original sample rate matches the new one, nothing happens
```
Comment (hajimehoshi) [inline audio/audio.go:442]: `If`

**Review by hajimehoshi — COMMENTED** [review]
Do we still need changes in audio/vorbis and audio/wav?

File: audio/audio.go (hunk 1 above)
Comment (hajimehoshi) [reply audio/audio.go:442]: s/nothing happens/Reample returns the source as it is/

File: audio/audio.go [hunk 2]
```diff
@@ -435,3 +436,13 @@ func (h *hookImpl) OnResumeAudio(f func() error) {
 func (h *hookImpl) AppendHookOnBeforeUpdate(f func() error) {
 	hooks.AppendHookOnBeforeUpdate(f)
 }
+
+// Resample explicitly resamples a stream to fit a new sample rate
```
Comment (hajimehoshi) [inline audio/audio.go:440]: Add `.` at the end of the sentence.

**Comment by RusAD32** [comment]
> Do we still need changes in audio/vorbis and audio/wav?

No, reverted them

**Review by hajimehoshi — APPROVED** [review]
LGTM, thanks!

**Pull request merged**